pkg/reconciler/v1alpha1/revision/cruds.go

```diff
@@ -85,8 +85,7 @@ func (c *Reconciler) checkAndUpdateService(ctx context.Context, rev *v1alpha1.Re
 	}
 	logger.Infof("Reconciling service diff (-desired, +observed): %v",
 		cmp.Diff(desiredService.Spec, service.Spec))
-	service.Spec = desiredService.Spec
```

Comment:
redundant assignment, the `desiredService` is originally copied from `service`, we can use it directly